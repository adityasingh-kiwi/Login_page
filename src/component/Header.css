.header {
    overflow: hidden;
    background-color:#363945;
      padding: 20px 10px;
      
  }

  .Background_image{
    background-image: url("/home/kiwitech/Desktop/Project/login_page/src/image/Survey.jpg");
    background-size: cover;
    overflow: hidden;
  }
  

  
  .header a {
    float: left;
    color: black;
    text-align: center;
    padding: 12px;
    text-decoration: none;
    font-size: 18px;
    line-height: 25px;
    border-radius: 4px;
  }
  
  .header a.logo {
    font-size: 25px;
    color: #fff;
    font-weight: bold;
  }
  
  .header a:hover {
    background-color: #ddd;
    color: black;
  }
  

  .header button.active {
    background-color: #2b3e50;
    color:rgb(0, 26, 255);
    border-radius: 18px;
    height: 70px;
    width: 200px;
  }
  .header button.logout {
    background-color: #003366;
    color: red;
    border-radius: 8px;
    height: 50px;
    width: 120px;
  }

  

  .header-right {
    float: right;

  }


  
  

 .profileImage{
   width: 50px !important;
   height: 50px !important;
 }

 .text_colour{
  color:white;
  cursor: pointer;

 }
 




              