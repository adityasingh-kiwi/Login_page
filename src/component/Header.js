import React, { useState, useEffect} from 'react'
import '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
import './Header.css';

import { useNavigate } from "react-router-dom";
import ModalOne from "./Modal"
import RequestService from "./api";


const Header = () => {

    const [getdata, setGetdata] = useState({email:" ",id:" "})


    const [modalState, handleModalState] = useState(false);
    const handleUserDetails = () => {

        handleModalState(true);

    }
    
  let Naviagte = useNavigate();
  

  const handleAdmin=()=>{
    Naviagte("/admin");
  }


    const fetchAllRequest = async ()=>{
        
        let res = await RequestService.getOne()
      
         let userEmail = localStorage.getItem("data")
         if(res.status == 200){
            console.log("userEmail------------->>>",userEmail);
          
             for(let property in res["data"]){
    

                if(res["data"][property]["userEmail"] == userEmail ){
                    
                   let update =  await RequestService.getOnlyOne(property);
                   
                    setGetdata(
                        {
                            ...getdata,
                            email:res["data"][property]["userEmail"],
                            id:property
                    } 
                    );
                }
              
             }
         }

        
    }
     useEffect(() => {
        fetchAllRequest();
    }, [])

     

    const handleOpenClose =(data)=>{
    

        if(!data){
            handleModalState(false);
        }
        
    }
    
    const handleLogout=()=>{
        Naviagte("/");

    }





    return (
        

        <div className="header">
            <a href="#default" className="logo">Naviagtion</a>

            <div className="header-right" style={{display:"flex"}}>
                
                <div className="col-md-4">

                    <img className="rounded-circle z-depth-2 profileImage" alt="20x20" src="https://mdbootstrap.com/img/Photos/Avatars/img%20(31).jpg"
                        data-holder-rendered="true"/>

                </div>
               
                <button className="active" onClick={handleUserDetails}>{getdata.email}</button>
                &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
                <button className="logout" onClick={handleLogout}>Logout</button>
                &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;

                {getdata.email==="admin@gmail.com" ?   <button className="logout"onClick={handleAdmin}>Admin</button> :null}
               

                <div className="background">
             



            </div>

            
            {
                <ModalOne  getdata={getdata}isopen={modalState} isOpenClose ={(data)=>handleOpenClose(data)}/>
            }
            <img className="mask-custom"></img>

        </div>
        </div>
    )
}



export default Header
