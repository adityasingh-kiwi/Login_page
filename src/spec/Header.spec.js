import { mount } from "enzyme";
import Header from "../component/Header";



describe('Header_page_test',()=>{
    it('userDeatils_button',()=>{
        let wrapper=shallow(<Header/>)
      const container = wrapper.find(".active")
       expect(container.length).toEqual(1)
     wrapper.find('.active').simulate('click');


    })
    it('Logout_button',()=>{
        let wrapper=shallow(<Header/>)
      const container = wrapper.find(".logout")
       expect(container.length).toEqual(1)
     wrapper.find('.logout').simulate('click');


    })
      it('Admin_button',()=>{
          let wrapper= mount(<Header/>)
        const container = wrapper.find("#admin")
      //    expect(container.length).toEqual(0);
      //  wrapper.find('#admin').simulate('click');


      })
     it('Snapshot_login',()=>{
      let wrapper=shallow(<Header/>)
      expect(wrapper).toMatchSnapshot();
      
  
    })

    it('userDeatils_button',()=>{
      let wrapper=shallow(<Header/>)
    const container = wrapper.find(".active");
   wrapper.find('.active').simulate('click');
   expect(
    wrapper.find('.active').prop('defaultValue'),
    ).toBeTruthy();


  })
  it('Logout_button',()=>{
    let wrapper=shallow(<Header/>)
    const container = wrapper.find(".logout")
       expect(container.length).toEqual(1)
     wrapper.find('.logout').simulate('click');
 expect(
  container.find('.logout').prop('defaultValue'),
  ).toBeFalsy();


})
    

})