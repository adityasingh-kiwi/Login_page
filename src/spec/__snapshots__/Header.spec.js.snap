// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header_page_test Snapshot_login 1`] = `
<div
  className="header"
>
  <a
    className="logo"
    href="#default"
  >
    Naviagtion
  </a>
  <div
    className="header-right"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      className="col-md-4"
    >
      <img
        alt="20x20"
        className="rounded-circle z-depth-2 profileImage"
        data-holder-rendered="true"
        src="https://mdbootstrap.com/img/Photos/Avatars/img%20(31).jpg"
      />
    </div>
    <button
      className="active"
      defaultValue={false}
      onClick={[Function]}
    >
       
    </button>
               
    <button
      className="logout"
      defaultValue={false}
      onClick={[Function]}
    >
      Logout
    </button>
               
    <div />
    <div
      className="background"
    />
    <ModalOne
      getdata={
        Object {
          "email": " ",
          "id": " ",
        }
      }
      isOpenClose={[Function]}
      isopen={false}
    />
    <img
      className="mask-custom"
    />
  </div>
</div>
`;
